src/xcm/mod.rs

```diff
@@ -277,27 +271,34 @@ mod tests {
 		);
 	}
 
-	#[test]
-	fn gas_to_weight() {
-		// https://docs.moonbeam.network/builders/interoperability/xcm/remote-evm-calls/#differences-regular-remote-evm
-		const WEIGHT_PER_GAS: u64 = 25_000;
-		assert_eq!(super::gas_to_weight(1), Weight::from_parts(WEIGHT_PER_GAS, 0));
-		const MAX_GAS_LIMIT: u64 = 720_000;
-		assert_eq!(super::gas_to_weight(MAX_GAS_LIMIT), Weight::from_parts(18_000_000_000, 0));
-	}
-
 	#[test]
 	fn transact() {
 		const GAS_LIMIT: u64 = 100_000;
-		let xt_weight = super::gas_to_weight(GAS_LIMIT) + DbWeight::get().reads(1);
-		let total_weight = super::weigh() + xt_weight;
+		let xt_weight =
+			<Test as crate::Config>::Weigher::transact(Parachain(EVM_PARA_ID), GAS_LIMIT);
+
+		let descend_origin = Weight::from_parts(5_992_000, 0);
```

Comment:
Should this not be using the Weigher as well?

Reply:
The Weigher implementation is in the runtime, so my view is that it would be the runtime that asserts for actual values. Fine to leave it as is tho. 👍